File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    result = []
    same, diff = 0, 0
    temp = s
    
    for i in range(len(temp)):
        if (s[0] == temp[i]):
            same += 1
        else:
            diff += 1
        
        if(same == diff):
            result.append(s[0:same+diff])
            s = temp[i+1:]
            same, diff = 0, 0
    if s != '':
        result.append(s)
        
    return len(result)
    

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    const result = [];
    let len = s.length;
    let temp = s;
    let same = 0;
    let diff = 0;
    
    for (let i = 0; i < len; i++) {
        if (s[0] == temp[i]) {
            same++;
        } else {
            diff++;
        }
        
        if (same === diff) {
            result.push(s.slice(0, same+diff));
            s = temp.slice(i+1, temp.length);
            same = 0;
            diff = 0;
        }
    }
    if (s !== '') {
        result.push(s);
    }
    
    return result.length
}